src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHelper.cs

```diff
@@ -0,0 +1,123 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http
+{
+    internal static class DiagnosticsHelper
+    {
+        internal static string GetRedactedUriString(Uri uri)
+        {
+            Debug.Assert(uri.IsAbsoluteUri);
+
+            if (GlobalHttpSettings.DiagnosticsHandler.DisableUriRedaction)
```

Comment:
> It looks like the code is treating removing the default port as one aspect of redaction.

I wouldn't say so, this is Uri-s default formatting strategy for both `uri.AbsoluteUri` and `uri.ToString()`, but in general you are right about removing information since this method removes Fragment for perforamance reasons.

Nevertheless, I would prefer to stick with the term *Redaction*, since it feels easier to communicate and Fragment doesn't seem to be worth picking something verbose/complicated like `TelemetryUsesUnmodifiedUri`. We can explain such details in docs.

Note: in #103769 we agreed to use a shared `System.Net.Http.DisableUriRedaction` switch for both HttpClientFactory logging redaction `uri.full` redaction.